### PR TITLE
Add Fastlane to automate deployments

### PR DESCRIPTION
**Pull request by kordianbruck**
Many helpful tools are bundled in this little piece of magic. 

We can also finally automate screenshots - still to do (https://docs.fastlane.tools/getting-started/android/screenshots/)

### CODE REVIEW

File: fastlane/Appfile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+json_key_file("/home/kb/Nextcloud/TUM/TCA/GooglePlayAndroidDeveloper-6abfb5ac51c5.json") # Path to the json secret file - Follow https://docs.fastlane.tools/actions/supply/#setup to get one
+package_name("de.tum.in.tumcampus") # e.g. com.krausefx.app
```
Comment (pfent) [inline fastlane/Appfile]: you might want to set up a environment variable for that?
and remove the example comments?

File: fastlane/report.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        
+      </testcase>
+    
+      
+      <testcase classname="fastlane.lanes" name="1: gradle" time="35.08299099">
+        
+      </testcase>
+    
+  </testsuite>
+</testsuites>
```
Comment (pfent) [inline fastlane/report.xml]: Maybe add that report to `.gitignore`?

**Comment by pfent** [comment]
Closing due to inactivity.

Feel free to open that PR up again, when there's actually "work in progress", but until then, please just keep your branch around.

**Comment by pfent** [comment]
@kordianbruck 👌👀👌👀👌👀👌👀👌👀 good shit go౦ԁ sHit👌 thats ✔ some good👌👌shit right👌👌there👌👌👌 right✔there ✔✔if i do ƽaү so my self 💯 i say so 💯 thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ💯 👌👌 👌НO0ОଠOOOOOОଠଠOoooᵒᵒᵒᵒᵒᵒᵒᵒᵒ👌 👌👌 👌 💯 👌 👀 👀 👀 👌👌Good shit

**Comment by psukys** [comment]
good shit

**Comment by pfent** [comment]
Google also just released an Android Publisher API:
https://android-developers.googleblog.com/2018/06/automating-your-app-releases-with.html

**Comment by kordianbruck** [comment]
@pfent can you have a look? I've released the latest beta with this.

File: fastlane/Fastfile [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        print_command: false,
+        properties: {
+            "android.injected.signing.store.file" => ENV["TCA_KEY_PATH"] + "/TCAkeystore.jks",
+            "android.injected.signing.store.password" => ENV["TCA_KEY_PW"],
+            "android.injected.signing.key.alias" => "key",
+            "android.injected.signing.key.password" => ENV["TCA_KEY_ALIAS_PW"],
+        }
+    )
+
+    upload_to_play_store(track: 'beta', skip_upload_screenshots: true, skip_upload_metadata: true, skip_upload_images: true)
```
Comment (pfent) [inline fastlane/Fastfile]: please add a parameter `apk`, pointing to the release apk

File: fastlane/Fastfile [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  end
+
+  desc "Deploy a new version to the Google Play"
+  lane :deploy do
+    gradle(task: "clean assembleRelease")
+    upload_to_play_store
```
Comment (pfent) [inline fastlane/Fastfile]: same same

**Comment by IzzySoft** [comment]
Could it be this nice stuff misses the per-release changelogs in fastlane? Last one there is for versionCode 522 from 2 years ago, the current one would be 558.

**Comment by kordianbruck** [comment]
Could be, yes

**Comment by IzzySoft** [comment]
Could you check (and possibly fix)? Would be great to have those up-to-date as well, so e.g. updates in the F-Droid app show what's new :wink:

**Comment by kordianbruck** [comment]
FDroid kicked us because of firebase anyway. So no way we are getting in there.

**Comment by IzzySoft** [comment]
Looks like you've [missed this](https://apt.izzysoft.de/fdroid/index/apk/de.tum.in.tumcampus) :wink: F-Droid is more than just "one repo", there [are many](https://android.izzysoft.de/articles/named/list-of-fdroid-repos).

But yes, Firebase is not FOSS. If you could replace that, it'd make the app more privacy-friendly even. Possible alternatives include MQTT, SSE (ServerSideEvents – that's what e.g. Tutanota uses), WebSockets (used by e.g. Telegram and Signal), XMPP… ([OpenPush](https://f-droid.org/en/2020/02/03/openpush-talk.html) is not ready yet, unfortunately). Using such an "open standard", it would even work on "GApp-less devices" (yes I know there's also microG and I'm even using it, but without enabling any Google stuff; I prefer to not send them any data).

**Pull request merged**